src/components/Task/MassActions/ConfirmationModal/MassActionsTasksConfirmationModal.tsx

```diff
@@ -20,23 +27,71 @@ export const MassActionsTasksConfirmationModal: React.FC<
 > = ({ open, idsCount, action, setOpen, onConfirm }) => {
   const { t } = useTranslation();
 
+  // KG: useState for checkbox
+  const [checked, setChecked] = useState(false);
+
+  const shouldConfirmDeletion = action === 'delete' && idsCount > 5;
+
+  const handleChange = (event) => {
+    setChecked(event.target.checked);
+  };
+
+  // KG: get correct task message based on action
+  function renderMessage() {
+    if (action === 'complete' || !shouldConfirmDeletion) {
+      return (
+        <Typography>
+          {t(
+            'Are you sure you wish to {{action}} the {{count}} selected tasks?',
+            { action, count: idsCount },
+          )}
+        </Typography>
+      );
+    } else {
```

Comment:
I think it would be best to move this render function into the JSX, to make it more readable.

We only use a render function like this in rare cases.